### ISSUE TITLE
Slightly underindented continuation should be accepted

### ISSUE THREAD

**Issue by ijackson** [rust-lang/rust-clippy#13001]
### Summary

This lint fires for any continuation that isn't indented as much as rustdoc thinks.  But, a one-character-less indent for a bulleted list is wholly unambiguous and doesn't pose the risks stated in the lint description.  It should be accepted.

### Lint Name

doc_lazy_continuation

### Reproducer

I tried this code:

```rust
/// List
///
///  * Item with some
///   continuation line that is not ambiguous 
pub fn item() {}
```

I saw this happen:

```
warning: doc list item missing indentation
 --> src/lib.rs:4:5
  |
4 | ///   continuation line that is not ambiguous 
  |     ^^
  |
  = help: if this is supposed to be its own paragraph, add a blank line
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_lazy_continuation
  = note: `#[warn(clippy::doc_lazy_continuation)]` on by default
help: indent this line
  |
4 | ///    continuation line that is not ambiguous 
  |       +

warning: `playground` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.47s
```

I expected to see this happen:

Program accepted without complaint.

### Version

```text
1.81.0-nightly (2024-06-24 6b0f4b5ec3aa707ecaa7)

From the playground.

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=8300c50ec4b1c6ffb70c1c1ebc08f3d0
```

**Comment by domenicquirl** [comment]
This just caused my entire workspace to light up on running `rustup update` because it also triggers for 

```rs
    /// This has the following conditions:
    ///   1. The first condition, which is described below.
    ///     - More detail about the condition.
    ///     - Additional information.
```

firing on each of the inner unordered list items because they are "only" indented by a single space, not two.

Imo linting on the original example is arguably fine, since at least it catches real inconsistencies (see the discussion in #13119). But here, the next line is in no way a continuation of the `1.` item's contents but is instead hierarchically a child / sub-item (i.e., as HTML this would be 

```html
This has the following conditions:
<ol>
  <li> The first condition, which is described below. 
    <ul>
      <li> More detail about the condition. </li>
      <li> Additional information. </li>
    </ul>
  </li>
</ol>
```

with the inner list being part of the first list item of the outer list).

I would argue that the lint should distinguish between text continuations and distinct, succeeding markdown elements and only apply to the former.

**Comment by ojeda** [comment]
> I expected to see this happen:
>
> Program accepted without complaint.

It depends on what you want. I would prefer it kept firing on your example. Perhaps it could be configurable.

**Comment by ojeda** [comment]
I am not sure what justification you are referring to (the current lint description seems fine to me), but even if it were wrong, the name of the lint is pretty clear about what it does.

As for the category, it seems to me "style" is a fairly suitable one, though I am not sure how those are picked. I guess you could argue for "pedantic" instead, but that is a different discussion.
